### PR TITLE
Post and comments

### PR DESCRIPTION
**Pull request by hedita**

### CODE REVIEW

File: src/Posts.jsx [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+  },
+];
+
+function Posts() {
+  const [isShow, setIsShow] = useState(true);
+  
+  const handleComments = () => {
+    setIsShow(!isShow);
+  }
+  const postsList = posts.map(({id,title,text,rate, image,comments},post) => {
```
Comment (takrami) [inline src/Posts.jsx]: never call index something else. here shouldn't called post.

File: src/Posts.jsx [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+  const commentList = comments.map(({author,text,likesCount},comment) =>{
+    return (
+      <Fragment key={comment}>
```
Comment (takrami) [inline src/Posts.jsx]: please don't use index as a key. you have id in the data. use id instead.

File: src/Posts.jsx [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+       <div style={{visibility: isShow ? "hidden" : "visible"}}>
+         <p key={comment}>{author}:{text}</p>
+         <p>likes: {likesCount}</p>
+       </div>
+      </Fragment> 
+    ) 
+  })
+    return (
+      <Fragment key={post}>
```
Comment (takrami) [inline src/Posts.jsx]: same.

File: src/Posts.jsx [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+];
+
+function Posts() {
+  const [isShow, setIsShow] = useState(true);
+  
+  const handleComments = () => {
+    setIsShow(!isShow);
+  }
+  const postsList = posts.map(({id,title,text,rate, image,comments},post) => {
+  const commentList = comments.map(({author,text,likesCount},comment) =>{
```
Comment (takrami) [inline src/Posts.jsx]: same

File: src/Posts.jsx (hunk 2 above)
Comment (hedita) [reply src/Posts.jsx]: Done.

File: src/Posts.jsx (hunk 3 above)
Comment (hedita) [reply src/Posts.jsx]: Done.

File: src/Posts.jsx (hunk 4 above)
Comment (hedita) [reply src/Posts.jsx]: Done.

File: src/Posts.jsx (hunk 1 above)
Comment (hedita) [reply src/Posts.jsx]: Done.

File: src/Posts.jsx [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+        text: "Wooow.",
+        likesCount: 2,
+      },
+    ],
+  },
+];
+
+function Posts() {
+  const [isShow, setIsShow] = useState(false);
+
```
Comment (takrami) [inline src/Posts.jsx]: either put an extra line after each or not. try to be consistent in your code format.

**Review by takrami — APPROVED** [review]
Beautiful! :star_struck: in my opinion, this part is well done and can be merged. but please add some styling. keep it simple. you do not need a lot of styles just some minimal styles can make it pretty.

File: src/Posts.jsx (hunk 5 above)
Comment (hedita) [reply src/Posts.jsx]: Done.

File: src/Posts.jsx [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+function Posts() {
+  const [isShow, setIsShow] = useState(false);
+  const handleComments = () => {
+    setIsShow(!isShow);
+  }
+  const postsList = posts.map(({id,title,text,rate, image,comments}) => {
+  const commentList = comments.map(({author,text,likesCount,id},comment) =>{
+    return (
+      <Fragment key={id}>
+       <div style={{visibility: isShow ? "visible" : "hidden"}}>
```
Comment (blacksrc) [inline src/Posts.jsx]: You should not render the comments when they are hidden.

File: src/Posts.jsx [hunk 7]
```diff
@@ -0,0 +1,94 @@
+import React, { Fragment,useState } from "react";
+
+const posts = [
```
Comment (blacksrc) [inline src/Posts.jsx]: Move this to another file and call it `postsData.js` and import it here.

File: src/Posts.jsx [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+        avatar: "https://ui-avatars.com/api/?name=SH",
+        text: "Wooow.",
+        likesCount: 2,
+      },
+    ],
+  },
+];
+
+function Posts() {
+  const [isShow, setIsShow] = useState(false);
```
Comment (blacksrc) [inline src/Posts.jsx]: What `isShow` referring to? From what I can infer it's a state in posts. So, I'll relate this to Posts component. Naming is very important. I suggest to rename it to `isCommentsVisible` or something similar that explains the relation.

File: src/Posts.jsx [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+  const handleComments = () => {
```
Comment (blacksrc) [inline src/Posts.jsx]: `handleComments` also is not a good naming. Something like: `handleCommentVisibility` would be better.

**Review by blacksrc — CHANGES_REQUESTED** [review]
Please fix the code format before sending the PR for the review.

File: src/Posts.jsx (hunk 7 above)
Comment (hedita) [reply src/Posts.jsx]: Done.

File: src/Posts.jsx (hunk 8 above)
Comment (hedita) [reply src/Posts.jsx]: Done.

File: src/Posts.jsx (hunk 9 above)
Comment (hedita) [reply src/Posts.jsx]: Done.

**Review by blacksrc — APPROVED** [review]

**Pull request merged**